MdeModulePkg/Universal/Variable/UefiVariablePolicy/ReadMe.md

```diff
@@ -0,0 +1,354 @@
+---
+title:      UEFI Variable Policy Whitepaper
+version:    1.0 RFC
+copyright:  Copyright (c) Microsoft Corporation.
+---
+
+# UEFI Variable Policy
+
+## Summary
+
+UEFI Variable Policy spec aims to describe the DXE protocol interface
+which allows enforcing certain rules on certain UEFI variables. The
+protocol allows communication with the Variable Policy Engine which
+performs the policy enforcement.
+
+The Variable Policy is comprised of a set of policy entries which
+describe, per UEFI variable (identified by namespace GUID and variable
+name) the following rules:
+
+-   Required variable attributes
+-   Prohibited variable attributes
+-   Minimum variable size
+-   Maximum variable size
+-   Locking:
+    -   Locking "immediately"
+    -   Locking on creation
+    -   Locking based on a state of another variable
+
+The spec assumes that the Variable Policy Engine runs in a trusted
+enclave, potentially off the main CPU that runs UEFI. For that reason,
+it is assumed that the Variable Policy Engine has no concept of UEFI
+events, and that the communication from the DXE driver to the trusted
+enclave is proprietary.
+
+At power-on, the Variable Policy Engine is:
+
+-   Enabled -- present policy entries are evaluated on variable access
+    calls.
+-   Unlocked -- new policy entries can be registered.
+
+Policy is expected to be clear on power-on. Policy is volatile and not
+preserved across system reset.
+
+## DXE Protocol Functions
```

Comment:
might be nice to call this DXE Protocol
Then have the structure and guid and global name

Then define the functions.  This makes skimming easier/faster.

Reply:
Fine with that.